Documentation/proposals/202407-remote-write.md

```diff
@@ -0,0 +1,128 @@
+# RemoteWrite CRD
+
+- Owners:
+  - [@superbrothers](https://github.com/superbrothers)
+- Related Tickets:
+  - [#6508](https://github.com/prometheus-operator/prometheus-operator/issues/6508)
+- Other docs:
+  - n/a
+
+## TL;DR
+
+This design doc proposes RemoteWrite CRD, which enables cluster admins to delegate the ability to configure Prometheus remote_rewrite configuration to application developers/operators.
+
+## Why
+
+The Prometheus remote_write configuration is defined in the Prometheus CRD. Currently, the configuration data generation is the responsibility of cluster admins.
+
+## Goals
+
+The main goal is to enable application developers/operators to self-service the remote write, and configure how the client sends metrics to the remote endpoint.
+
+This means exposing a new CRD to configure Prometheus remote_write configuration.
+
+## Non-goals
+
+We do not cover "RemoteRead" CRD here. We can still implement it later if needed.
```

Comment:
```suggestion
Prometheus has two APIs for integrating with remote storage: 'remote write' and 'remote read'. Since there has been no confirmed demand to treat remote read as a CRD, we do not cover "RemoteRead" CRD here. We can still implement it later if needed.
```